### PR TITLE
Week 3 - Step 6 - Adding deleting and limit display of comments functionality

### PR DESCRIPTION
**Pull request by malcolmrjones**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        Query queryAllComments = new Query("Comment");
+        PreparedQuery results = datastore.prepare(queryAllComments);
+
+        for (Entity comment : results.asIterable()) {
+            datastore.delete(comment.getKey());
+        }
+
+    }
+
+}
```
Comment (ccyang314) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: nit: add a new line at the end of this file. This is will make git diff works well if the future changes contains content under this block.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -44,15 +44,20 @@ public void init() {
...
     Query queryAllComments = new Query("Comment").addSort("time", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(queryAllComments);
     
+
     for (Entity comment : results.asIterable()) {
       String commentContent = (String) comment.getProperty("content");
       comments.add(commentContent);
     }
+
+    //Prevent comment count to be over the total number of comments
```
Comment (ccyang314) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: add a space between `//` and `Prevent`

**Review by ccyang314 — APPROVED** [review]
LGTM.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -45,7 +45,8 @@ function displayGreeting(greetingText) {
  * Makes request to receive all comments from server
  */
 function fetchComments() {
-  const commentsPromise = fetch("/data?comment-count=3");
+  const commentsPromise = fetch("/data?" + "comment-count=" + document.getElementById("comment-count").value)
```
Comment (brucezheng) [inline portfolio/src/main/webapp/script.js]: Nit: add semicolon

**Review by brucezheng — APPROVED** [review]

**Pull request merged**